transforms/templateDirectives/invalidTemplateAttributes.js

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { walkParse5Ast, serializeParse5Ast } from '../parse5Utils'
+
+const rootTemplateDirectives = new Set(['lwc:preserve-comments', 'lwc:render-mode'])
+const nonRootTemplateDirectives = new Set([
+  'for:each',
+  'for:index',
+  'if:true',
+  'if:false',
+  'lwc:if',
+  'lwc:elseif',
+  'lwc:else'
```

Comment:
Maybe we can import this list directly from `@lwc/template-compiler`? I'm pretty sure there is a list already in there like this.